### PR TITLE
Wu2018 Group 2

### PR DESCRIPTION
**Pull request by SShivang**
Just pushed a draft summary.  Need to make some changes.

### CODE REVIEW

**Comment by sritank** [comment]
Have you pushed your final summary? The one here seems short

**Comment by SShivang** [comment]
I have added some more changes. @sritank Please take a look.

File: _summaries/wu2018group_2.md [hunk 1]
```diff
@@ -1,19 +1,51 @@
...
-* What interesting variants are explored?
+Core Idea:
+
+The batch norm introduced an optimization in which the outputs of the layers were normalized for a particular batch (scaled and recentered). This lead to significant improvement in performance for a lot of deep learning tasks (one of the most important improvements). However with smaller batch sizes, this technique doesn't work as well. This may be the case if it is impossible to have large batch sizes (models such as video classification models) and you have to train on a smaller batch size. This paper introduces a group norm which works on smaller batches and has consistent improvements of performance across batch sizes. Group norm breaks down does normalization of the input layer by breaking channels into groups.
+
+Technical Realization:
+
+![Performance of Batch Norm vs Group Norm](wu2018group_2_1.png)
+
+Batch norms are highly dependent on batches. However, for tasks such as image segmentation, object detection, and video classification, models train on smaller batch sizes as their inputs are so large. In this case batch norm is less useful.
```
Comment (sritank) [inline _summaries/wu2018group_2.md:18]: I think this repeats the core idea. You could remove it

File: _summaries/wu2018group_2.md [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
+
+![All normalization techniques](wu2018group_2_6.png)
+
+ This paper takes inspiration from techniques such as layer norms and instance norms, which are batch independent. However, Instance Norm goes to far as it doesn't leverage all the information across channels. Therefore the it loses too much statisitcal properties that other techniques employ. In relation to layer norm, group norm does better since the model has the flexibility of learning different distribution for each group and doesn't assume that the statisitcal distribution accross all channels is the same (in case of layer norm). This leads to improvements in representations. Group Normalization does better than all of the techniques in this respect as it is batch independent, and it has a good trade-off between how many channels it normalizes across (exploiting channel dependence while still not assuming all the channels have same distributional characterisitcs).
```
Comment (sritank) [inline _summaries/wu2018group_2.md:22]: I feel like you should go more into the technical realization here. An important point is that they do not apply GN directly on the image inputs, but in some cases, after the convolution process. They apply GN on the features generated after the convolution layer in several experiments in the paper. 
You could discuss why GN works better in the performance section

File: _summaries/wu2018group_2.md [hunk 3]
```diff
@@ -1,19 +1,51 @@
...
+![All normalization techniques on ResNet-18](wu2018group_2_5.png)
+
+The Group Norm has similar performance to batch norm, however for larger batch sizes it doesn't do as well as batch norm. In the figure above we see that for large batch sizes, batch norm does pretty well, however as the
+batch size decreases, Batch Norm's performance deteriorates. On the other hand, group norm does pretty well
+across all batches.
+
+![Performance improvement on object segmentation](wu2018group_2_3.png)
+
+Group Norm outperforms Batch Norm on the task of object segmentation as object segmentation networks train with smaller batch sizes. This is because object segmentation demands higher resolution input so batch sizes tend to be much smaller. Here we see that Group Norm outperforms Batch Norm. Note that BN is frozen during finetuning as this gave better results.
+
```
Comment (sritank) [inline _summaries/wu2018group_2.md:35]: I believe they did an additional experiment using GN to train Mask R-CNN from scratch. They got better result than training it with BN and no pre-training.

File: _summaries/wu2018group_2.md (hunk 1 above)
Comment (SShivang) [reply _summaries/wu2018group_2.md:18]: Moved it to earlier section

File: _summaries/wu2018group_2.md (hunk 3 above)
Comment (SShivang) [reply _summaries/wu2018group_2.md:35]: Added to the section.

File: _summaries/wu2018group_2.md (hunk 2 above)
Comment (SShivang) [reply _summaries/wu2018group_2.md:22]: Added that realization to this section.

**Comment by SShivang** [comment]
@zhaoyue-zephyrus I have done all the necessary changes.

**Comment by zhaoyue-zephyrus** [comment]
@SShivang Thanks! I'm merging it
